API.md

```diff
@@ -0,0 +1,123 @@
+# Authmosphere API documentation
+
+## Mock tooling
+
+This tooling provides an abstraction to easily mock oAuth 2.0 (RFC) related endpoints.
+
+It helps to easily write integration tests without writing extensive boilerplate code and without disabling oAuth for tests.
+
+This tooling is based on Nock, a HTTP mocking library. For more information about Nock see [Nock documentation](https://github.com/node-nock/nock).
+
+### mockAccessTokenEndpoint
+
+Creates a __very basic__ mock of token endpoint as defined in [RFC 6749](https://tools.ietf.org/html/rfc6749).
+
+The mocked endpoint will return a token(Link to type) with the scopes specified in the request.
+
+* ⚠️ The mock does not validate the request
+* ⚠️ The mock holds a state that contains the created tokens
+* ⚠️ `cleanMock` resets the state and removes all nocks
+
+#### Usage
+
+```typescript
+mockAccessTokenEndpoint({
+  url: 'http://some.oauth.endpoint/access_token',
+  times: 1
+});
+```
+
+#### Signature
+
+`mockAccessTokenEndpoint(options) => nock.Scope`
+
+#### Arguments
+
+* `options`:
```

Comment:
we have a type for `options`, right? shall we name it here and link it maybe?

Reply:
Nevertheless I think it is good to mention the properties explicitly here!

Reply:
Added a link for all `options` ;)